base/path.jl

```diff
@@ -517,12 +517,20 @@ function relpath(path::String, startpath::String = ".")
     curdir = "."
     pardir = ".."
     path == startpath && return curdir
-    path_drive, path_without_drive = splitdrive(path)
-    startpath_drive, startpath_without_drive = splitdrive(startpath)
-    path_arr  = split(abspath(path_without_drive),      path_separator_re)
-    start_arr = split(abspath(startpath_without_drive), path_separator_re)
     if Sys.iswindows()
-        lowercase(path_drive) != lowercase(startpath_drive) && return abspath(path)
+        path_drive, path_without_drive = splitdrive(path)
+        startpath_drive, startpath_without_drive = splitdrive(startpath)
+        isempty(startpath_drive) && (startpath_drive = path_drive) # by default assume same as path drive
+
+        path_drive = uppercase(path_drive) # canonicalize drive letters to uppercasing for comparison
+        startpath_drive = uppercase(startpath_drive)
+
+        path_drive != startpath_drive && return abspath(path) # if drives differ return first path
+        path_arr  = split(abspath(path_drive * path_without_drive),      path_separator_re)
+        start_arr = split(abspath(startpath_drive * startpath_without_drive), path_separator_re)
```

Comment:
Since `path_drive` may involve a DNS lookup, I think we should preserve the input case:
```suggestion
        startpath_drive = uppercase(startpath_drive) # canonicalize drive letters to uppercasing for comparison
        uppercase(path_drive) == startpath_drive || return abspath(path) # if drives differ return first path
        path_arr  = split(abspath(path_drive * path_without_drive),      path_separator_re)
        start_arr = split(abspath(path_drive * startpath_without_drive), path_separator_re)
```

Reply:
Even better, thank you for the review. PR updated.